### ISSUE TITLE
IPrincipal Interface support multi assumeRoleAction

### ISSUE THREAD

**Issue by muramasa64** [aws/aws-cdk#6699]
The type of the property assumeRoleAction of the IPrincipal interface is string.

Now that ABAC is supported, assumeRoleAction must be a string[].

### Use Case

The following AWS Security Blog, which describes ABAC, contains the Role Trust Policy required for using ABAC.

https://aws.amazon.com/jp/blogs/security/rely-employee-attributes-from-corporate-directory-create-fine-grained-permissions-aws/

The Role Trust Policy is quoted below. The Action defined in the Statement has multiple entries.

```json
{
	"Version": "2012-10-17",
	"Statement": [
		{
			"Effect": "Allow",
			"Principal": {
				"Federated": "arn:aws:iam::999999999999:saml-provider/ExampleCorpProvider"
			},
			"Action": [
				"sts:AssumeRoleWithSAML",
				"sts:TagSession"
			],
			"Condition": {
				"StringEquals": {
					"SAML:aud": "https://signin.aws.amazon.com/saml"
				},
				"StringLike": {
					"aws:RequestTag/project": "*",
					"aws:RequestTag/jobfunction": [
						"SystemsEngineer",
						"DatabaseEngineer"
					]
				}
			}
		}
	]
}

```

### Proposed Solution

Change the type of assumeRoleAction property of IPrincipal to string[].

### Other

* [ ] :wave: I may be able to implement this feature request
* [x] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

**Comment by redbaron** [comment]
Looks like duplicate of https://github.com/aws/aws-cdk/issues/2041

**Comment by lkr2des** [comment]
Seems similar issue.
I need to create a AssumeRolePolicyDocument  like below attached to a federated role, unable to do that today.Any ideas or help is appreciate.

{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Principal": {
        "Federated": "arn:aws:iam::xxxxxxxxxxxxxxxxx:saml-provider/okta-quicksight"
      },
      "Action": "sts:AssumeRoleWithSAML",
      "Condition": {
        "StringEquals": {
          "SAML:aud": "https://signin.aws.amazon.com/saml"
        }
      }
    },
    {
      "Effect": "Allow",
      "Principal": {
        "Federated": "arn:aws:iam::xxxxxxxxx:saml-provider/okta-quicksight"
      },
      "Action": "sts:TagSession",
      "Condition": {
        "StringLike": {
          "aws:RequestTag/Email": "*"
        }
      }
    }
  ]
}

**Comment by atpollmann** [comment]
This is resolved with the `withSessionTags` method as described here https://github.com/aws/aws-cdk/pull/17689

**Comment by comcalvi** [comment]
Resolved by #17689
